### PR TITLE
feat: [CDS-78126]: Onboarded GitX settings for services

### PR DESCRIPTION
**Pull request by sunnykesh-harness**
### Summary
Onboarding the gitX settings for the remote services.

Changes are : 

- Making GitX settings label generic ( removing entity types as suggested by PM)
- Onboarding DefaultStoreType and GitXEnforced settings  for service creation from dashboard, listing, pipeline. We need to handle these 2 via parent as need to have proper storeType Inline/Remote preselected in 1st view itself.
- Default Connectors and Repo setting will be taken care by GitSync Form itself so we do not need to handle via parent.
- Fixed scroll overlap issue with create modal content.

Use case tested : 
Changes are just to pre-populate and disable the storeType for remote services so do not have much impact. 
Service creation for different combinations of settings are tested as shown in screen recording

#### Screenshots

https://github.com/harness/harness-core-ui/assets/63278928/5b229327-6fd5-406f-a234-cdb8b5d40192

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Review by praneshg239 — APPROVED** [review]

**Comment by praneshg239** [comment]
Approved behalf of CV

File: src/modules/27-platform/default-settings/utils/utils.ts [hunk 1]
```diff
@@ -30,3 +30,19 @@ export const getDefaultStoreType = (
...
+ * @returns
+ */
+export function getDefaultStoreTypeFromSettings(
+  gitXSetting?: ResponseListSettingResponseDTO | null
+): StoreMetadata['storeType'] {
+  return gitXSetting
+    ? getSettingValue(gitXSetting, SettingType.ENFORCE_GIT_EXPERIENCE) === 'true'
+      ? StoreType.REMOTE
+      : getDefaultStoreType(gitXSetting)
+    : StoreType.INLINE
```
Comment (ishantmehtaharness) [inline src/modules/27-platform/default-settings/utils/utils.ts:47]: looks like a complicated ternary, do we want to simplify it?

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/BaseDeployServiceEntity.tsx [hunk 2]
```diff
@@ -52,6 +54,9 @@ import { FormMultiTypeCheckboxField } from '@common/components/MultiTypeCheckbox
 import type { Error, ServiceResponseDTO } from 'services/cd-ng'
 import { ErrorHandler } from '@modules/10-common/components/ErrorHandler/ErrorHandler'
 import { StoreType } from '@modules/10-common/constants/GitSyncTypes'
+import useRBACError from '@modules/20-rbac/utils/useRBACError/useRBACError'
+import { getDefaultStoreTypeFromSettings, getSettingValue } from '@modules/27-platform/default-settings/utils/utils'
+import { SettingType } from '@modules/10-common/constants/Utils'
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/BaseDeployServiceEntity.tsx:59]: Isnt eslint failing here, `@modules/10-common/` should be imported before `@modules/27-platform/`?

**Review by ishantmehtaharness — APPROVED** [review]
looks good to me, with few nit

File: src/modules/27-platform/default-settings/utils/utils.ts (hunk 1 above)
Comment (sunnykesh-harness) [reply src/modules/27-platform/default-settings/utils/utils.ts:47]: I think not so complicated as it is having just 2 ifs. Also, kept this as separate utility function and have explained in code comment above it.

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/BaseDeployServiceEntity.tsx (hunk 2 above)
Comment (sunnykesh-harness) [reply src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/BaseDeployServiceEntity.tsx:59]: Yes, it should. Looks like lint rules related to control imports are not running from sometime. Will have to check it is on purpose or due to any issue.

**Review by sonaligoyal-harness — APPROVED** [review]

**Pull request merged**